src/device.jl

```diff
@@ -79,8 +79,10 @@ mutable struct Device
         end
         this = new(dev_ptr)
         finalizer(this) do this
-            this != C_NULL && SoapySDRDevice_unmake(this)
-            this = Ptr{SoapySDRDevice}(C_NULL)
+            if this.ptr != C_NULL
+                SoapySDRDevice_unmake(this)
+                this.ptr = Ptr{SoapySDRDevice}(C_NULL)
+            end
```

Comment:
```suggestion
            isopen(this) || return
            SoapySDRDevice_unmake(this)
            this.ptr = Ptr{SoapySDRDevice}(C_NULL)
```

to make it similar with the Stream version.

Reply:
Thanks!
Fixed in https://github.com/JuliaTelecom/SoapySDR.jl/compare/8dca16d67b4d998154724f0a4f32f86596931e4c..5e28e1072187dee5e71c6a8f8838db347418e856